src/components/CharacterCard.vue

```diff
@@ -0,0 +1,93 @@
+<template>
+  <div class="character-card">
+    <div class="character-card__image-wrapper">
+      <img class="character-card__image" :src="image" :alt="name" />
+    </div>
+    <div class="character-card__desc">
+      <h3 class="character-card__name">{{ name }}</h3>
+      <p class="character-card__info">{{ `${gender}, ${species}` }}</p>
+      <p class="character-card__info--type">Type: {{ type || 'Unknown' }}</p>
+    </div>
+    <div class="character-card__status" :class="`character-card__status--${status.toLowerCase()}`">
+      {{ status }}
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import type { Character } from '@/services/characters'
+
+const props = defineProps<{ characterData: Character }>()
+
+const { gender, image, name, species, status, type } = props.characterData
+</script>
+
+<style scoped lang="scss">
+.character-card {
+  $max-card-width: 320px;
+  $max-image-height: 300px;
+
+  position: relative;
+  display: flex;
+  flex-direction: column;
+  gap: $space-xs;
+  overflow: hidden;
+  width: 100%;
+  height: 100%;
+  max-width: $max-card-width;
+  border: 1px solid $border;
+  border-radius: $radius;
+  background-color: $card;
+  cursor: pointer;
+
+  &__image-wrapper {
+    max-height: $max-image-height;
+  }
+
+  &__image {
+    height: $max-image-height;
+    width: 100%;
+    object-fit: cover;
+  }
+
+  &__desc {
+    display: flex;
+    flex-direction: column;
+    gap: $space-xs;
+    padding: $p-xs $p-sm;
+
+    & * {
```

Comment:
pls read about css performance 

> The key to optimizing CSS selectors is to focus on the rightmost selector, also called the key selector (coincidence?). Here’s a much more expensive selector: A.class0007 * {}. Although this selector might look simpler, it’s more expensive for the browser to match. Because the browser moves right to left, it starts by checking all the elements that match the key selector, “*“. This means the browser must try to match this selector against all elements in the page.

On page like that it doesn't really matter, but you should avoid this practice.